event/consume.go

```diff
@@ -0,0 +1,112 @@
+package event
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ONSdigital/dp-import-cantabular-dataset/schema"
+	importapi "github.com/ONSdigital/dp-import-api/models"
+	"github.com/ONSdigital/dp-api-clients-go/dataset"
+
+	kafka "github.com/ONSdigital/dp-kafka/v2"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// Consume converts messages to event instances, and pass the event to the provided handler.
+func (p *Processor) Consume(ctx context.Context, cg kafka.IConsumerGroup, h Handler) {
+	// consume loop, to be executed by each worker
+	var consume = func(workerID int) {
+		for {
+			select {
+			case msg, ok := <-cg.Channels().Upstream:
+				if !ok {
+					log.Info(ctx, "upstream channel closed - closing event consumer loop", log.Data{"worker_id": workerID})
+					return
+				}
+
+				ctx := context.Background()
```

Comment:
Would it make sense to make this ctx as a child context of the Consume method's passed in context ?
Since parent context closure could potentially propagate down.

Reply:
Yeah I guess so